hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1300,6 +1300,79 @@ private class Emit(
             res.isNull))
 
         EmitTriplet(setup, m, res.invoke[Double]("doubleValue"))
+      case x@MakeNDArray(dataIR, shapeIR, rowMajorIR) =>
+        val dataContainer = dataIR.pType.asInstanceOf[PArray]
+        val shapePType = shapeIR.pType.asInstanceOf[PTuple]
+        val nDims = shapePType.size
+
+        val datat = emit(dataIR)
+        val shapet = emit(shapeIR)
+        val rowMajort = emit(rowMajorIR)
+
+        val t = x.pType.asInstanceOf[PNDArray]
+        val srvb = new StagedRegionValueBuilder(mb, t.representation)
+
+        val setup = Code(
+          shapet.setup,
+          datat.setup,
+          rowMajort.setup)
+        val value = coerce[Long](Code(
+          srvb.start(),
+          srvb.addInt(0),
+          srvb.advance(),
+          srvb.addInt(0),
+          srvb.advance(),
+          shapet.m.mux(
+            Code._fatal("Missing shape"),
+            Code(
+              srvb.addBaseStruct(t.representation.fieldType("shape").asInstanceOf[PBaseStruct], { srvb: StagedRegionValueBuilder =>
+                Code(
+                  srvb.start(),
+                  {
+                    var index = 0
+                    var shapeCopyingCode = Code._empty[Unit]
+                    while (index < nDims) {
+                      shapeCopyingCode = Code(shapeCopyingCode,
+                        shapePType.isFieldMissing(coerce[Long](shapet.v), index).mux[Unit](
+                          Code._fatal(s"shape missing at index $index"),
+                          Code(srvb.addLong(region.loadLong(shapePType.loadField(coerce[Long](shapet.v), index))), srvb.advance())
+                        ))
+                      index += 1
+                    }
+                    shapeCopyingCode
+                  }
+                )
+              }),
+              srvb.advance(),
+
+              srvb.addBaseStruct(t.representation.fieldType("strides").asInstanceOf[PBaseStruct], { srvb =>
+                Code (
+                  srvb.start(),
+                  {
+                    var index = 0
+                    var strideWritingCode = Code._empty[Unit]
+                    while (index < nDims) {
+                      strideWritingCode = Code(strideWritingCode,
+                        srvb.addLong(dataContainer.elementType.byteSize), srvb.advance())
```

Comment:
This doesn't seem right. The strides are all the `elementType` size? I think the ith stride should be `elementType.byteSize` * shape[i+1] * ... * shape[nDims].